src/reason_pprint_ast.ml

```diff
@@ -697,6 +699,16 @@ let rules = [
   [
     (TokenPrecedence, (fun s -> (Nonassoc, s = "=>")));
   ];
+  [
+    (TokenPrecedence, (fun s ->
+      let is_infix_op = Hashtbl.mem infix_operators s in
```

Comment:
Maybe useful, but how do we allow user to choose the associativity? AFAIK Haskell uses a two-passes algorithm which determines associativity and precedence in the first pass.